### PR TITLE
Xmrig

### PR DESCRIPTION
**Pull request by 88plug**

### CODE REVIEW

File: monero/deploy.yaml [hunk 1]
```diff
@@ -3,40 +3,36 @@ version: "2.0"
 
 services:
   xmrig:
-    image: andrey01/xmrig:latest
+    image: cryptoandcoffee/akash-xmrig:1
```
Comment (boz) [inline monero/deploy.yaml:6]: it'd be good to have the source of this be available.

File: monero/run.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+#function build(){
+apt-get update && \
+        apt install iputils-ping msr-tools kmod git build-essential libbz2-dev cmake libuv1-dev libssl-dev libhwloc-dev wget gcc g++ -y && \
+        apt clean && \
+        rm -rf /var/lib/apt/lists/*
+
+git clone https://github.com/xmrig/xmrig.git ; cd xmrig
```
Comment (boz) [inline monero/run.sh:8]: `--depth 1` would make this faster.

File: monero/run.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+sed -i 's/constexpr const int kDefaultDonateLevel = 1;/constexpr const int kDefaultDonateLevel = '${DONATE_LEVEL}';/g' src/donate.h
+sed -i 's/constexpr const int kMinimumDonateLevel = 1;/constexpr const int kMinimumDonateLevel = '${DONATE_LEVEL}';/g' src/donate.h
+
+mkdir build ; cd build ; cmake .. -DCMAKE_C_COMPILER=gcc-11 -DCMAKE_CXX_COMPILER=g++-11 -DCMAKE_BUILD_TYPE=Release ; make -j$(nproc) && cp xmrig /usr/bin
+#}
+
+if [[ $POOL == *"hashvault"* ]]; then
+echo "Checking nearest Hashvault location"
+hashvault_locations=(pool.hashvault.pro:443 chicago01.hashvault.pro:443 frankfurt01.hashvault.pro:443 moscow01.hashvault.pro:443 singapore01.hashvault.pro:443)
```
Comment (boz) [inline monero/run.sh:17]: 1. indent conditionals
2. use newlines to make array more readable and editable.

File: monero/run.sh [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+for url in "${hashvault_locations[@]}"
+do
+   :
+   echo "Checking average ping of $url for 10 seconds"
+   original_url=$url
+   url=${url%:*}
+   ping=$(ping -c 10 $url | tail -1| awk '{print $4}' | cut -d '/' -f 2)
+   url=$original_url
+   echo "$url,$ping" >> speed.log
```
Comment (boz) [inline monero/run.sh:27]: you can do

```sh
echo -n "$url" >> speed.log
# other stuff...
echo "$ping" >> speed.log
```

**Review by boz — APPROVED** [review]
Looks great.  I'd recommend more readable shell formatting going forward.

File: monero/deploy.yaml (hunk 1 above)
Comment (88plug) [reply monero/deploy.yaml:6]: The source is the Dockerfile in the repo.
https://github.com/ovrclk/awesome-akash/blob/xmrig/monero/Dockerfile

**Pull request merged**